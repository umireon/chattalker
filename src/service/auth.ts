import type { AppContext } from '../../constants'
import type { Auth } from 'firebase/auth'
import { signInWithCustomToken } from 'firebase/auth'

interface AuthenticateWithTokenOptions {
  readonly token: string
  readonly uid: string
}

export const authenticateWithToken = async (auth: Auth, { authenticateWithTokenEndpoint }: AppContext, { token, uid }: AuthenticateWithTokenOptions) => {
  const query = new URLSearchParams({ token, uid })
  const response = await fetch(`${authenticateWithTokenEndpoint}?${query}`)
  if (!response.ok) throw new Error('Authentication failed')
  const customToken = await response.text()
  const credential = await signInWithCustomToken(auth, customToken)
  return credential
}
