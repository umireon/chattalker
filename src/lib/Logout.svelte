<script lang="ts">
  import { type Auth } from 'firebase/auth'

  export let auth: Auth

  async function handleLogout () {
    await auth.signOut()
    location.href = '/'
  }
</script>

<main>
  <p>
    <button type="button" on:click={handleLogout}>Logout</button>
  </p>
</main>
