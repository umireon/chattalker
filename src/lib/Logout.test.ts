import { type Auth } from "firebase/auth";
import Logout from "./Logout.svelte";
import { render } from "@testing-library/svelte";

test("Logout snapshot", () => {
  const props = { auth: {} as Auth };
  const component = render(Logout, { props });
  expect(component.container).toMatchSnapshot();
});
