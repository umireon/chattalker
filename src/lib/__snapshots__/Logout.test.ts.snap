// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logout snapshot 1`] = `
<body>
  <div>
    <main>
      <p>
        <button
          type="button"
        >
          Logout
        </button>
      </p>
    </main>
  </div>
</body>
`;
